rxnetty/src/main/java/io/reactivex/netty/protocol/http/UnicastContentSubject.java

```diff
@@ -138,7 +137,14 @@ private UnicastContentSubject(final State<T> state, long noSubscriptionTimeout,
      */
     public boolean disposeIfNotSubscribed() {
         if (state.casState(State.STATES.UNSUBSCRIBED, State.STATES.DISPOSED)) {
-            state.bufferedSubject.subscribe(Subscribers.empty()); // Drain all items so that ByteBuf gets released.
+            state.bufferedSubject.subscribe(new Action1<T>() {
+                @Override
+                public void call(T t) {
+                    if (t instanceof ReferenceCounted) {
```

Comment:
No, at this point it will have a refCnt of 2!

This is what I was wondering as well..the bytebuf is created with 1.. when it comes into this subject it is retained so it has a refcnt of 2. if noone picks it up releasing just once will not help since it will stick at 1.. I wonder if we should retain it at all?

If we don't retain it here:
- we can release() if noone subscribes
- it gets properly released if one subscribes and calls release()

Is there a specific reason it gets retained in the subject?